core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java

```diff
@@ -536,6 +536,15 @@ public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredComm
       Object key = command.getKey();
       MVCCEntry e = (MVCCEntry) ctx.lookupEntry(key);
       Metadata metadata = command.getMetadata();
+      // When it isn't the primary owner, just accept the removal as is, trusting the primary did the appropriate checks
+      if (!distributionManager.getCacheTopology().getSegmentDistribution(command.getSegment()).isPrimary()) {
```

Comment:
you can just check for `FlagBitSets.BACKUP_WRITE`. The primary owner sets it when sends the command to the backup owners.
```java
// Mark the command as a backup write so it can skip some checks
command.addFlags(FlagBitSets.BACKUP_WRITE);
```

Reply:
Ah, let me swap that out.